samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/BeginDeliveryActionListener.java

```diff
@@ -453,10 +455,22 @@ else if (minute > 1) {
     		if (extTimeService.hasExtendedTime() && extTimeService.getTimeLimit() > 0) {
     			control.setTimeLimit(extTimeService.getTimeLimit());
     		}
-    		String timeLimitInSetting = control.getTimeLimit() == null ? "0" : control.getTimeLimit().toString();
     		Date attemptDate = unSubmittedAssessmentGrading.getAttemptDate();
+    		long timeLimitInSetting = control.getTimeLimit();
+    		Long now = new Date().getTime();
+    		Long start = attemptDate.getTime();
+    		if((now - start) > (timeLimitInSetting*1000)) {
+    			// check if the queue is ahead and already submitted it
+    			TimedAssessmentQueue queue = TimedAssessmentQueue.getInstance();
+    			TimedAssessmentGradingModel timedAG = (TimedAssessmentGradingModel)queue.
+    					get(unSubmittedAssessmentGrading.getAssessmentGradingId());
+    			// if it was submitted (race condition) while checking, unblock it - sam will synch soon
+    			if(timedAG != null && !timedAG.getSubmittedForGrade()) {
+    				delivery.setTimeExpired(true);
+    			}
+    		}
     		delivery.setBeginTime(attemptDate);
-    		String timeBeforeDueRetract = delivery.getTimeBeforeDueRetract(timeLimitInSetting);
+    		String timeBeforeDueRetract = delivery.getTimeBeforeDueRetract(control.getTimeLimit() == null ? "0" : String.valueOf(control.getTimeLimit()));
```

Comment:
Hey Brian,

And adding the overhead of a lib call that does the exact same thing?
On Jun 6, 2016 8:41 AM, "Brian Jones" <notifications@github.com> wrote:

> In
> samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/BeginDeliveryActionListener.java
> <https://github.com/sakaiproject/sakai/pull/2755#discussion_r65882652>:
>
> >      		delivery.setBeginTime(attemptDate);
> > -    		String timeBeforeDueRetract = delivery.getTimeBeforeDueRetract(timeLimitInSetting);
> > +    		String timeBeforeDueRetract = delivery.getTimeBeforeDueRetract(control.getTimeLimit() == null ? "0" : String.valueOf(control.getTimeLimit()));
>
> You can use NumberUtils.toInt() from Apache Commons to avoid this ternary
> operator.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/sakaiproject/sakai/pull/2755/files/be7b76fbc2dad2497f949c6dda1ea2ea6f6d12e9#r65882652>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AEnF-LFlxmburQX45zU4dajT2Dgcz73_ks5qJBVdgaJpZM4It5Ue>
> .
>